ImmuDB4Net/ImmuClient.cs

```diff
@@ -94,37 +90,99 @@ internal IConnection Connection
+    /// </summary>
+    /// <value>Default: true</value>
+    public bool DeploymentInfoCheck
     {
-        return new ImmuClientBuilder();
+        get
+        {
+            return stateHolder.DeploymentInfoCheck;
+        }
+
+        set
+        {
+
+            stateHolder.DeploymentInfoCheck = value;
+        }
     }
+    /// <summary>
+    /// Gets or sets the length of time the <see cref="ImmuClient.Close" /> function is allowed to block before it completes.
+    /// </summary>
+    /// <value>Default: 2 sec</value>
+    public TimeSpan ConnectionShutdownTimeout { get; set; }
+
+    /// <summary>
+    /// Gets or sets the length of time interval between periodic checks of idle connections to be closed. 
+    /// </summary>
+    /// <value></value>
+    public TimeSpan IdleConnectionCheckInterval { get; internal set; }
 
-    public class LibraryWideSettings
+    /// <summary>
+    /// Gets the grpc address of the ImmuDB server
+    /// </summary>
+    /// <value></value>
+    /// <remarks>
+    /// This value is computed from the server url and server port arguments that come either from 
+    /// <see cref="ImmuClientBuilder.Build()" /> or <see cref="ImmuClient(string, int)" /> constructor
+    /// </remarks>
+    public string GrpcAddress { get; }
+
+    /// <summary>
+    /// Gets the <see cref="LibraryWideSettings" /> object with the process wide settings
+    /// </summary>
+    /// <value></value>
+    public static LibraryWideSettings GlobalSettings
     {
-        public int MaxConnectionsPerServer { get; set; }
-        public TimeSpan TerminateIdleConnectionTimeout { get; set; }
-        public TimeSpan IdleConnectionCheckInterval { get; set; }
-        internal LibraryWideSettings()
+        get
         {
-            //these are the default values
-            MaxConnectionsPerServer = 2;
-            TerminateIdleConnectionTimeout = TimeSpan.FromSeconds(60);
-            IdleConnectionCheckInterval = TimeSpan.FromSeconds(6);
+            return globalSettings;
         }
     }
 
+    /// <summary>
+    /// Creates a new instance of <see cref="ImmuClientBuilder" /> factory object for <see cref="ImmuClient" /> instances
+    /// </summary>
+    /// <returns>A new instance of <see cref="ImmuClient" /></returns>
+    public static ImmuClientBuilder NewBuilder()
+    {
+        return new ImmuClientBuilder();
+    }
+
+    /// <summary>
+    /// Initializes a new instance of <see cref="ImmuClient" />. It uses the default value of 'localhost:3322' as ImmuDB server address and port and 'defaultdb' as database name.
+    /// </summary>
+    /// <returns></returns>
     public ImmuClient() : this(NewBuilder())
     {
 
     }
 
+    /// <summary>
+    ///  Initializes a new instance of <see cref="ImmuClient" />. It uses the implicit 'defaultdb' value for the database name.
+    /// </summary>
+    /// <param name="serverUrl">The ImmuDB server address, ex: localhost or http://localhost </param>
```

Comment:
```suggestion
    /// <param name="serverUrl">The ImmuDB server address, e.g. localhost or http://localhost </param>
```

(http://www.jvhighschool.com/study-tip-how-to-use-i-e-e-g-and-ex-properly/#:~:text=E.g.%20is%20another%20Latin%20abbreviations,actually%20stands%20for%20“exercise.”)

Reply:
ok